packages/protocol-parser/src/padSliceHex.ts

```diff
@@ -0,0 +1,8 @@
+import { Hex } from "viem";
+
+export function padSliceHex(data: Hex, start: number, end?: number): Hex {
```

Comment:
should we add a small typedoc comment with something like "slices the hex string while padding data out of bound data with `0`; including start index, excluding end index"?

Reply:
done! also renamed to `readHex` because I didn't like `padSliceHex`